### ISSUE TITLE
cannot deploy cc-ui properly

### ISSUE THREAD

**Issue by HirossxD** [linkedin/cruise-control-ui#115]
i have tried to download latest release - [cruise-control-ui-0.4.0.tar.gz](https://github.com/linkedin/cruise-control-ui/releases/download/v0.4.0/cruise-control-ui-0.4.0.tar.gz) and nothing happened. only http://localhost:9090/kafkacruisecontrol worked. http://localhost:9090/ got jetty 404 error.
I tried to setup nginx and in config use root /opt/cruise-control/cruise-control-ui/dist but this only showed me blank page with title.
I tried to download Source code (tar.gz) or clone directly from main branch and run ```npm install``` and ```npm run dev``` it got error because in ```build/webpack.dev.conf.js``` was undefined path. so i added line ```var path = require('path')``` to that file.
afterwards after running npm run dev or ```npm run start``` app started on port 8090 but with following error:

```
# npm run start

> cruise-control-ui@0.3.0 start /opt/cruise-control/cruise-control-ui
> test -d cruise-control-ui && cd cruise-control-ui; node build/dev-server.js

[HPM] Proxy created: /kafkacruisecontrol  -> http://localhost:9090
> Starting dev server...
{ parser: "babylon" } is deprecated; we now treat it as { parser: "babel" }.


 DONE  Compiled successfully in 7530ms                                                                                                            1:39:56 PM
> Listening at http://0.0.0.0:8090

(node:21239) UnhandledPromiseRejectionWarning: Error: Exited with code 3
    at ChildProcess.cp.once.code (/opt/cruise-control/cruise-control-ui/node_modules/opn/index.js:85:13)
    at Object.onceWrapper (events.js:286:20)
    at ChildProcess.emit (events.js:198:13)
    at maybeClose (internal/child_process.js:982:16)
    at Socket.stream.socket.on (internal/child_process.js:389:11)
    at Socket.emit (events.js:198:13)
    at Pipe._handle.close (net.js:607:12)
(node:21239) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 1)
(node:21239) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
```

after this website is reachable under http://1.2.3.4:8090/ , but shows only blank page (as is in index.html).

i can see following error in the console (i am using nginx reverse proxy on port 8080 which wil proxypass ip of cc with port 8090)

```
GET
http://1.2.3.4:8080/app.js
[HTTP/1.1 404 Not Found 51ms]

Loading failed for the <script> with source “http://1.2.3.4:8080/app.js”. cc:10:44
```

btw i have functional CC setup and running.
```
# curl http://localhost:9090/kafkacruisecontrol/state
MonitorState: {state: RUNNING(20.000% trained), NumValidWindows: (5/5) (100.000%), NumValidPartitions: 76/76 (100.000%), flawedPartitions: 0}
ExecutorState: {state: NO_TASK_IN_PROGRESS}
AnalyzerState: {isProposalReady: true, readyGoals: [NetworkInboundUsageDistributionGoal, CpuUsageDistributionGoal, PotentialNwOutGoal, LeaderReplicaDistributionGoal, NetworkInboundCapacityGoal, LeaderBytesInDistributionGoal, DiskCapacityGoal, ReplicaDistributionGoal, RackAwareGoal, TopicReplicaDistributionGoal, NetworkOutboundCapacityGoal, CpuCapacityGoal, DiskUsageDistributionGoal, NetworkOutboundUsageDistributionGoal, ReplicaCapacityGoal]}
AnomalyDetectorState: {selfHealingEnabled:[], selfHealingDisabled:[DISK_FAILURE, BROKER_FAILURE, GOAL_VIOLATION, METRIC_ANOMALY, TOPIC_ANOMALY, MAINTENANCE_EVENT], selfHealingEnabledRatio:{DISK_FAILURE=0.0, BROKER_FAILURE=0.0, GOAL_VIOLATION=0.0, METRIC_ANOMALY=0.0, TOPIC_ANOMALY=0.0, MAINTENANCE_EVENT=0.0}, recentGoalViolations:[], recentBrokerFailures:[], recentMetricAnomalies:[], recentDiskFailures:[], recentTopicAnomalies:[], recentMaintenanceEvents:[], metrics:{meanTimeBetweenAnomalies:{GOAL_VIOLATION:0.00 milliseconds, BROKER_FAILURE:0.00 milliseconds, METRIC_ANOMALY:0.00 milliseconds, DISK_FAILURE:0.00 milliseconds, TOPIC_ANOMALY:0.00 milliseconds}, meanTimeToStartFix:0.00 milliseconds, numSelfHealingStarted:0, numSelfHealingFailedToStart:0, ongoingAnomalyDuration=0.00 milliseconds}, ongoingSelfHealingAnomaly:None, balancednessScore:100.000}
```

Could you please help me resolve this issue ? If you need any more info just ask...

**Comment by HirossxD** [comment]
allright so issue was with the nginx reverse proxy configuration , however after running ```npm run start``` error persists.
I am able to see the CCFE website now, however, i cannot see the existing cluster of metrics.

i can also see the following error in the html console
```
http://108.142.123.19:8080/static/config.csv?_t=1691485246540
[HTTP/1.1 404 Not Found 56ms]
```

so probably CCFE doesnt know backend endpoint as it cannot parse config.csv because it is not retrieving it.

**Comment by HirossxD** [comment]
I have tried to ignore everything in .eslintignore , did not help.
However, after running ```npm run build``` and setting up ```nginx``` , website now works! (before i had issue with nginx reverse proxy on jump server so page was blank).
I have also tried to wget https://github.com/linkedin/cruise-control-ui/releases/download/v0.4.0/cruise-control-ui-0.4.0.tar.gz
and with local nginx this works as well !

However it would be nice to resolve ```npm run start``` error.
